File: src/main/kotlin/onl/tesseract/srp/skill/adapter/serverside/task/CraftTickRunnable.java
package onl.tesseract.srp.skill.adapter.serverside.task;

import onl.tesseract.srp.skill.domain.model.crafting.CraftTask;
import org.bukkit.scheduler.BukkitRunnable;

import java.time.Duration;

public class CraftTickRunnable extends BukkitRunnable {
    private CraftTask craftTask;

    public CraftTickRunnable(CraftTask craftTask) {
        this.craftTask = craftTask;
    }

    @Override
    public void run() {
        craftTask = craftTask.tick(Duration.ofSeconds(1));
    }
}

File: src/main/kotlin/onl/tesseract/srp/infrastructure/runnable/skill/CraftTickRunnable.kt
package onl.tesseract.srp.infrastructure.runnable.skill

import onl.tesseract.srp.domain.skill.crafting.CraftTask
import org.bukkit.scheduler.BukkitRunnable
import kotlin.time.Duration.Companion.seconds

class CraftTickRunnable(
    private val craftTask: CraftTask
) : BukkitRunnable() {

    override fun run() {
        craftTask.tick(1.seconds)
    }
}
